packages/atomic/src/components/common/facets/facet-tabs/facet-tabs-utils.ts

```diff
@@ -0,0 +1,41 @@
+import {CategoryFacet, DateFacet, Facet, NumericFacet} from '@coveo/headless';
+import {shouldDisplayOnCurrentTab} from '../../../../utils/tab-utils';
+
+type AnyFacetType = Facet | NumericFacet | CategoryFacet | DateFacet;
+
+/**
+ * Updates the visibility of a facet based on the active tab.
+ * @param tabsIncluded - An array of tab names that should include the facet.
+ * @param tabsExcluded - An array of tab names that should exclude the facet.
+ * @param activeTab - The currently active tab.
+ * @param facet - The facet to update.
+ */
+export function updateFacetVisibilityForActiveTab(
+  tabsIncluded: string[] | string,
+  tabsExcluded: string[] | string,
+  activeTab: string,
+  facet?: AnyFacetType
+): void {
+  if (
+    !facet ||
+    (tabsIncluded.length === 0 && tabsExcluded.length === 0) ||
```

Comment:
Right now that will only happen if someone does something like `tabs-excluded='"[]"'` instead of `tabs-excluded="[]"` since all the components that use this function use @Arrayprop for the tabs-excluded/tabs-included properties. 

@fbeaudoincoveo should I just remove the possibility of providing a single string as a value for tabsExcluded/tabsIncluded? It would prevent this problem from happening at all..

Reply:
Nevermind, @ArrayProp needs either an array or a stringified array, not a single string value. Found a better way to handle this:

```typescript
   updateFacetVisibilityForActiveTab(
      [...this.tabsIncluded],
      [...this.tabsExcluded],
      this.tabManagerState?.activeTab,
      this.facet
    );
```
and updateFacetVisibilityForActiveTab now requires a string array